### PR TITLE
fix: `make verify-kube-connect` broken on arm64 architectures. #12617

### LINKED ISSUE #12617: `make verify-kube-connect` broken on arm64 architectures.
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

I want to compile the argocd binaries on my ubuntu ARM64 server, according to the [local development guide](https://argo-cd.readthedocs.io/en/stable/developer-guide/toolchain-guide/#build-or-pull-the-required-docker-image).

I went according to the [local development guide](https://argo-cd.readthedocs.io/en/stable/developer-guide/toolchain-guide/#build-or-pull-the-required-docker-image).

**To Reproduce**

1. Clone the `argoproj/argo-cd` repo to an ARM64 ubuntu server (assuming go, git, docker are available).
2. Run `make verify-kube-connect`.

**Expected behavior**

I expect an output of the kubectl version.

**Screenshots**

**Version**

Less relevant as this issue is related to contribution to the project.

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Step 29/34 : COPY --from=node /usr/local/bin/node /usr/local/bin
 ---> Using cache
 ---> f9ef6bf61c45
Step 30/34 : COPY --from=node /opt/yarn-v1.22.4 /opt/yarn-v1.22.4
 ---> Using cache
 ---> 1c66766592b5
Step 31/34 : COPY ./test/container/entrypoint.sh /usr/local/bin
 ---> Using cache
 ---> ee4f19d4a452
Step 32/34 : ARG UID
 ---> Using cache
 ---> 73d38066f1b9
Step 33/34 : RUN useradd -l -u ${UID} -d /home/user -s /bin/bash user &&     echo "user ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/user &&     mkdir -p /home/user/.kube &&     chown -R user /home/user &&     chgrp -R user /home/user &&     HOME=/home/user git config --global user.name "ArgoCD Test User" &&     HOME=/home/user git config --global user.email "noreply@example.com" &&     mkdir -p /go/pkg &&     mkdir -p /var/run/sshd &&     mkdir -p /root/.ssh &&     mkdir -p /go &&     chown root /etc/ssh/ssh_host_*_key* &&     chmod 0600 /etc/ssh/ssh_host_*_key &&     mkdir -p /tmp/go-build-cache &&     mkdir -p /home/user/.cache &&     ln -s /usr/local/bin/node /usr/local/bin/nodejs &&     ln -s /usr/local/lib/node_modules/npm/bin/npm-cli.js /usr/local/bin/npm &&     ln -s /usr/local/lib/node_modules/npm/bin/npx-cli.js /usr/local/bin/npx &&     ln -s /opt/yarn-v1.22.4/bin/yarn /usr/local/bin/yarn &&     ln -s /opt/yarn-v1.22.4/bin/yarnpkg /usr/local/bin/yarnpkg &&     mkdir -p /var/lib/registry
 ---> Using cache
 ---> d7b4841c9d31
Step 34/34 : ENTRYPOINT ["/usr/local/bin/entrypoint.sh"]
 ---> Using cache
 ---> aaa52b1e8725
Successfully built aaa52b1e8725
Successfully tagged argocd-test-tools:latest
docker tag argocd-test-tools argocd-test-tools:latest
docker run --rm -it --name argocd-test-client -u 1000:1000 -e HOME=/home/user -e GOPATH=/go -e ARGOCD_E2E_K3S= -e GITHUB_TOKEN -e GOCACHE=/tmp/go-build-cache -e ARGOCD_LINT_GOGC=20 -v "/home/adam/go/src:/go/src:delegated" -v /home/adam/go/pkg/mod:/go/pkg/mod:delegated -v /home/adam/.cache/go-build:/tmp/go-build-cache:delegated -v /home/adam/.kube:/home/user/.kube:delegated -v /tmp:/tmp:delegated -w /go/src/github.com/argoproj/argo-cd argocd-test-tools:latest bash -c "kubectl version"
bash: line 1: /usr/local/bin/kubectl: cannot execute binary file: Exec format error
```

### PR DESCRIPTION
**Pull request by adamiBs**
Replace the `bitnami/kubectl` version in order to support both `amd64` along with `arm64` system architectures.
This is done according to [this issue](https://github.com/bitnami/containers/issues/19130) in the [bitnami/containers](https://github.com/bitnami/containers) repository.
The [image I propose using](https://hub.docker.com/layers/bitnami/kubectl/1.26/images/sha256-17fa84851a8b07abd4db95661eb0d3af0f14a5831f415a3a8b533c1ac5dbfc77?context=explore) supports both amd64 and arm64, instead of just amd64.

Fixes #12617.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [X] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by 34fathombelow** [comment]
lgtm! I'm totally cool with this change.  We are actually in the process of pinning our container images by digest hashes which are immutable.  If you would like to include this it would be much appreciated.   I would recommend using a tool such as [crane](https://github.com/google/go-containerregistry/tree/main/cmd/crane) to retrieve the image digest.
 ```bash
crane digest docker.io/bitnami/kubectl:1.26
sha256:625467eb8c3a3d60232923404941c32e787eb9003e644d0fa8258b0efa7f6a7f
```
The pinned image would look like this:
`docker.io/bitnami/kubectl:1.26@sha256:625467eb8c3a3d60232923404941c32e787eb9003e644d0fa8258b0efa7f6a7f`

Note: An image digest is different than a digest from the manifest which are platform specific and mutable. 
```bash
crane manifest docker.io/bitnami/kubectl:1.26 | jq
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
  "manifests": [
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "size": 429,
      "digest": "sha256:17fa84851a8b07abd4db95661eb0d3af0f14a5831f415a3a8b533c1ac5dbfc77",
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "size": 429,
      "digest": "sha256:4144f4b1e2085c57ea2970d7b73b5cd3969b87a5a853af96a1f6031e790ee186",
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    }
  ]
}
```

**Comment by adamiBs** [comment]
Thanks for the review and the additional context. I've performed the change now.
Along with that, I will create an issue along with a PR changing the other images in the Dockerfile, as a change like that would be out of the context of this PR and issue.

File: test/container/Dockerfile
Comment (34fathombelow) [inline test/container/Dockerfile:14]:
```suggestion
FROM docker.io/bitnami/kubectldocker.io/bitnami/kubectl:1.26@sha256:625467eb8c3a3d60232923404941c32e787eb9003e644d0fa8258b0efa7f6a7f as kubectl
```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: test/container/Dockerfile
Comment (adamiBs) [reply test/container/Dockerfile]: Cool!
I wasn't familiar with that format. Will take into account when implementing the changes related to issue #12622.
I've overidden the existing commit with your proposal.

Comment (34fathombelow) [reply test/container/Dockerfile]: my apologies, my copy pasta was insufficient this morning.  I corrected the suggested changes.

Comment (adamiBs) [reply test/container/Dockerfile]: I noticed after responding =)

**Review by crenshaw-dev — APPROVED** [review]
lgtm, @34fathombelow I’ll wait for your review before merging. Thanks all!

**Review by 34fathombelow — APPROVED** [review]
Lgtm thank you for your contribution

**Pull request merged**
